tests/rptest/perf/ts_write_openmessaging_test.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.cluster import cluster
+from rptest.services.openmessaging_benchmark import OpenMessagingBenchmark
+from ducktape.mark import parametrize
+from rptest.services.redpanda import SISettings
+
+
+class TSWriteOpenmessagingTest(RedpandaTest):
+    BENCHMARK_WAIT_TIME_MIN = 10
+
+    def __init__(self, ctx):
+        si_settings = SISettings(test_context=ctx)
+        self._ctx = ctx
```

Comment:
Thanks for putting this together! Some other settings that may be worth setting:

- local retention, which can be set via `alter_topic()` or as a cluster config `retention_local_target_bytes_default` and `retention_local_target_ms_default`
- `cloud_storage_cache_chunk_size` to something lower than the `SISettings` default segment size or 16MB (or setting the segment size to something larger)

I think these would at least give a solid starting point of testing the "normal" cloud storage read path, ensuring we hit remote storage and consume with chunked readers.

I'm not sure it makes sense to do in the context of this PR just yet, but it would be interesting to also test spillover, using some low value for `cloud_storage_spillover_manifest_max_segments`.

Reply:
I've added a TS read test which modifies `retention_local_target_bytes_default` to force TS reads in the test. `cloud_storage_cache_chunk_size` is 16MB for both tests.

For spillover would we want a separate test for it? Or maybe we should add it to the TS read test. Beyond that is there a metric I can use to validate spillover is occurring?

Reply:
Thanks!

Sorry there was a typo in my comment, I meant we should set the chunk size to be lower than the segment size _of_ 16MB, or raise the segment size. So either set `log_segment_size` to some 32MB or set `cloud_storage_cache_chunk_size` to e.g. 1MB. If the intent was that 16MB is actually lower than 16MiB, then perhaps add a comment explaining the relation that we want our chunks smaller than the segment size to exercise chunked reads.

>For spillover would we want a separate test for it?

I think it'd be fine to keep it in the same test case. It doesn't really detract from the performance characteristics of the read path, other than incurring an additional manifest download. I.e., much of the read path is the same when using spillover; there's just some extra logic to account for additional new downloads.

> Or maybe we should add it to the TS read test. Beyond that is there a metric I can use to validate spillover is occurring?

The uploaded spillover metric will be useful here.
https://github.com/redpanda-data/redpanda/blob/2400a6de63e37c85a726148ed8563ebbd19e48c1/src/v/cloud_storage/remote_probe.cc#L124-L131

Reply:
Should `SISettings` in ducktape should be updated to include `cloud_storage_cache_chunk_size`?

Reply:
Yea probably, but I don't feel strongly that you need to do that here. Feel free to just use `extra_rp_conf`

Reply:
Maybe it makes sense to use segment/chunk size similar to what we have in our cloud (128/16)? This test is not running on every PR so we can probably afford a workload that runs for 20 or 30 minutes. Or maybe even more. 
I'd also like to propose adding another OMB configuration. First configuration should be similar to what you added but longer and with more load. And the second configuration should have large number of partitions and "wide" reads. For instance, 5000 partitions per topic.

Few other things to consider:
- spillover should be enabled by default at this point. It makes sense to decrease the size of the spillover manifest by setting `cloud_storage_spillover_manifest_max_segments` to 80 (this will trigger spillover after every 10GiB of data uploaded to S3)
- The test should saturate our storage limits so we will end up uploading and deleting data simultaneously. So, for instance, if the test produces 100GiB per partition we can set `retention.bytes` to 20GiB.
- All these numbers scale with number of partitions so they should be adjusted for the configuration with wide reads. And we probably want to tweak number of subscriptions and number of consumers per subscription to make sure that the individual fetch requests are fetching from large number of partitions in parallel. I don't really understand how subscriptions and consumers are mapping to consumer groups so it's difficult for me to say how exactly they should be changed.

Reply:
@Lazin : Maybe we let this PR go in as is and we have a second PR with the proposed TS-specific stuff we want to test (and you can help with that second PR?)

Reply:
@Lazin I've updated this PR to introduce the following;
- `log_segment_size` will be 16MB
- `retention_local_target_bytes_default` will be 16MB
- `cloud_storage_cache_chunk_size` will be 8MB
- `cloud_storage_spillover_manifest_max_segments` will be 10

For a couple of minutes in the 15min test this will trigger simultaneous reads and writes to cloud storage. 

I agree with your other points. However, we're trying to cap the runtime of these tests to ~15mins to allow us to run all of them on a nightly basis. The tests are mainly to ensure that there are no regressions in common hot paths of our codebase. And running them frequently allows us to minimize what changes we have to examine when a regression does occur.

We'll also be writing a set of regression tests that will run on a weekly or bi-weekly basis. These tests won't have caps on their runtimes and I think the test you purposed will be much more suited for that set of tests.